core/models/configuration_template.rb

```diff
@@ -4,30 +4,30 @@
 
 # The representation of the template file that provides tools to get information out of it
 class ConfigurationTemplate
-  attr_reader :template_type
+  attr_reader :template_type, :node_configurations
```

Comment:
You can not provide attribute reader for all the nodes.

If you need nodes, then provide iterator for them, but do not publish this array.